src/main/java/service/impl/CsvReportGeneratorService.java

```diff
@@ -0,0 +1,30 @@
+package service.impl;
+
+import db.ShopStorage;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+import service.ReportGeneratorService;
+
+public class CsvReportGeneratorService implements ReportGeneratorService {
+    private final ShopStorage fruitStorage;
```

Comment:
This class doesn't follow single responsibility principle.
It doing both: generating report and writing report to file.
Please split into 2 classes: one will generate report (and result will be String or List<String> or String[], what is more convenient for you), second one will accept this report and filename to just write it